ElectricityBillApi.Tests/ElectricProviderTests.cs

```diff
@@ -0,0 +1,94 @@
+﻿using ElectricityBillApi.Models;
+using System;
+using Xunit;
+
+namespace ElectricityBillApi.Tests
+{
+    public class ElectricProviderTests
+    {
+        private ElectricProvider _sut;
+
+        public ElectricProviderTests()
+        {
+            _sut = new ElectricProvider();
+        }
+
+        [Fact]
+        public void CalculatePrice_CalculatesCorrectPrice()
+        {
+            // arrange
+            var loc1 = new Location()
+            {
+                X = 0,
+                Y = 0,
+                Z = 0
+            };
+            var loc2 = new Location()
+            {
+                X = 1,
+                Y = 1,
+                Z = 1
+            };
+            decimal plantPrice = 5;
+            var plant = new PowerPlant()
+            {
+                Location = loc1,
+                ElectricityPrice = plantPrice
+            };
+
+            // act
+            _sut.Subscribe(plant);
+            var actual = _sut.CalculatePrice(new Address() { Location = loc2 });
+            var expected = 3 * plantPrice;
+
+            // assert
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public void CalculatePrice_ThrowsIfNoPlantIsSubscribed()
+        {
+            // arrange
+
+            var loc2 = new Location()
+            {
+                X = 1,
+                Y = 1,
+                Z = 1
+            };
+
+            // act
+
+            // assert
+            Assert.ThrowsAny<ArgumentNullException>(() => _sut.CalculatePrice(new Address() { Location = loc2 }));
+        }
+
+        [Fact]
+        public void Subscribe_SubscribeTwoTimesThrows()
```

Comment:
try to use supplementary words to help identify the different parts of tests. Also segregate the different parts through `_`.
For example:
`Subscribe_WhenAlreadSubscribed_ThrowsInvalidOperationException`

Also specify the exception you throw for clarity.

Reply:
Renamed them.